CreditManagement/views.py

```diff
@@ -22,62 +22,71 @@ def get_queryset(self):
+        """
+        Checks whether user has access
+        :param request:
+        :return: None if user has access, otherwise a HTTPResponse object
+        """
         return None
 
 
-class TransactionAlterView(CustomAccessMixin, TemplateView):
+class TransactionBaseAlterView(CustomAccessMixin, TemplateView):
+    """
+    Provides a base view for the transactions which automatically alters the layout depending on whether
+    a Pending Transaction model is created or altered.
+    """
     template_name = None
     template_add_name = "credit_management/transaction_add.html"
     template_edit_name = "credit_management/transaction_edit.html"
-    transaction = None
+    transaction_obj = None
 
     def dispatch(self, request, *args, **kwargs):
         transaction_id = request.GET.get('id', None)
         if transaction_id is not None:
-            self.transaction = get_object_or_404(PendingTransaction, pk=transaction_id)
-        return super(TransactionAlterView, self).dispatch(request, *args, **kwargs)
-
-    def get_transaction(self):
-        transaction_id = self.request.GET.get('id', None)
-        try:
-            return PendingTransaction.objects.get(pk=transaction_id)
-        except PendingTransaction.DoesNotExist:
-            return None
-
-    def get_form(self, transaction=None, data=None):
+            self.transaction_obj = get_object_or_404(PendingTransaction, pk=transaction_id)
+        return super(TransactionBaseAlterView, self).dispatch(request, *args, **kwargs)
+
+    def get_form(self, data=None):
+        """
+        Returns the form the view is supposed to display
+        :param data: request.POST data
+        :return:
+        """
         raise NotImplementedError()
 
     def get_context_data(self, *args, **kwargs):
-        context = super(TransactionAlterView, self).get_context_data(*args, **kwargs)
+        context = super(TransactionBaseAlterView, self).get_context_data(*args, **kwargs)
 
-        transaction_obj = self.get_transaction()
         # get the form
-        context['form'] = self.get_form(transaction=transaction_obj)
+        context['form'] = self.get_form()
 
         context['redirect'] = self.request.GET.get('redirect', reverse('index'))
         context['chain'] = self.request.GET.get('chain', False)
 
         # Set the correct template
-        if transaction_obj is None:
+        if self.transaction_obj is None:
             self.template_name = self.template_add_name
         else:
             self.template_name = self.template_edit_name
         return context
 
     def post(self, request, *args, **kwargs):
-        transaction_obj = self.get_transaction()
-        form = self.get_form(transaction=transaction_obj, data=request.POST)
+        form = self.get_form(transaction=self.transaction_obj, data=request.POST)
 
         if form.is_valid():
             form.save()
-            if transaction_obj is None:
+            if self.transaction_obj is None:
```

Comment:
Using different views for creation and updating of transactions makes the code much cleaner (no need for `if transaction is None` in a lot of places). See e.g. the Django built in generic editing views.

Reply:
Come to think of it, there's no need whatsoever to allow transactions to be altered. It can lead to really confusing situations for users.... E.g. someone on the receiving end would expect that his balance is fixed and cannot go down without interaction by himself. With this implementation that's however not the case (if I understand correctly) as a transaction sender might later alter a transaction.

It seems often that you think that more functionality is better for the user. It's really not. You can find a lot online about what constitutes good UX, e.g. https://www.atlanticbt.com/insights/functionality_vs_usability_ux/. I often need to argue a lot about this while for me it's usually pretty clear, and it's getting annoying.